central/graphql/resolvers/node_vulnerabilities.go

```diff
@@ -80,8 +82,366 @@ func (resolver *Resolver) NodeVulnCounter(ctx context.Context, args RawQuery) (*
+	unFixableCVEs, err := vulnLoader.FromQuery(ctx, unFixableVulnsQuery)
+	if err != nil {
+		return nil, err
+	}
+
+	return mapNodeCVEsToVulnerabilityCounter(fixableVulns, unFixableCVEs), nil
+}
+
+// CreatedAt is the time a node CVE first seen in the system
+func (resolver *nodeCVEResolver) CreatedAt(ctx context.Context) (*graphql.Time, error) {
+	value := resolver.data.GetCveBaseInfo().GetCreatedAt()
+	return timestamp(value)
+}
+
+// CVE name of the node CVE
+func (resolver *nodeCVEResolver) CVE(ctx context.Context) string {
+	return resolver.data.GetCveBaseInfo().GetCve()
+}
+
+// EnvImpact is the fraction of nodes that contain the nodeCVE
+func (resolver *nodeCVEResolver) EnvImpact(ctx context.Context) (float64, error) {
+	n, d, err := resolver.getEnvImpactComponentsForNodeCVE(ctx)
+	if err != nil {
+		return 0, err
+	}
+	if d == 0 {
+		return 0, nil
+	}
+	return float64(n) / float64(d), nil
+}
+
+func (resolver *nodeCVEResolver) getEnvImpactComponentsForNodeCVE(ctx context.Context) (numerator, denominator int, err error) {
+	allNodesCount, err := resolver.root.NodeGlobalDataStore.CountAllNodes(ctx)
+	if err != nil {
+		return 0, 0, err
+	}
+	if allNodesCount == 0 {
+		return 0, 0, nil
+	}
+	nodeLoader, err := loaders.GetNodeLoader(ctx)
+	if err != nil {
+		return 0, 0, err
+	}
+	withThisCVECount, err := nodeLoader.CountFromQuery(resolver.withNodeVulnerabilityScope(ctx), search.EmptyQuery())
+	if err != nil {
+		return 0, 0, err
+	}
+	return int(withThisCVECount), allNodesCount, nil
+}
+
+func (resolver *nodeCVEResolver) withNodeVulnerabilityScope(ctx context.Context) context.Context {
+	return scoped.Context(ctx, scoped.Scope{
+		ID:    resolver.data.GetId(),
+		Level: v1.SearchCategory_NODE_VULNERABILITIES,
+	})
+}
+
+// FixedByVersion returns the version of the parent component that removes this CVE
+func (resolver *nodeCVEResolver) FixedByVersion(ctx context.Context) (string, error) {
+	return resolver.getNodeComponentFixedByVersion(ctx)
+}
+
+func (resolver *nodeCVEResolver) getNodeComponentFixedByVersion(_ context.Context) (string, error) {
+	scope, hasScope := scoped.GetScope(resolver.ctx)
+	if !hasScope {
+		return "", nil
+	}
+	if scope.Level != v1.SearchCategory_NODE_COMPONENTS {
+		return "", nil
+	}
+	edgeID := edges.EdgeID{ParentID: scope.ID, ChildID: resolver.data.GetId()}.ToString()
+	edge, found, err := resolver.root.NodeComponentCVEEdgeDataStore.Get(resolver.ctx, edgeID)
+	if err != nil || !found {
+		return "", err
+	}
+	return edge.GetFixedBy(), nil
+}
+
+// ID of the node CVE
+func (resolver *nodeCVEResolver) ID(ctx context.Context) graphql.ID {
+	value := resolver.data.GetId()
+	return graphql.ID(value)
+}
+
+// IsFixable returns whether node CVE is fixable by any component
+func (resolver *nodeCVEResolver) IsFixable(ctx context.Context, args RawQuery) (bool, error) {
+	// TODO : Why do we remove this field query and then add it back again in addScopeContextOrNodeCVEQuery ?
```

Comment:
@md2119

Reply:
This is I think because if this subresolver is invoked with scoping from a component/node/cluster, we define the `IsFixable` as if it is fixable in any context, not just the context that it's being queried in